internal/pkg/cli/app_init.go

```diff
@@ -0,0 +1,231 @@
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/prompt"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// AppInitOpts holds the configuration needed to create a new application.
+type AppInitOpts struct {
+	// Fields with matching flags.
+	AppType        string
+	AppName        string
+	DockerfilePath string
+
+	// Injected fields by parent commands.
+	projectName string
+
+	// Interfaces to interact with dependencies.
+	fs             afero.Fs
+	manifestWriter archer.ManifestIO
+	prompt         prompter
+}
+
+// Ask prompts for fields that are required but not passed in.
+func (opts *AppInitOpts) Ask() error {
+	if opts.AppType == "" {
+		if err := opts.askAppType(); err != nil {
+			return err
+		}
+	}
+	if opts.AppName == "" {
+		if err := opts.askAppName(); err != nil {
+			return err
+		}
+	}
+	if opts.DockerfilePath == "" {
+		if err := opts.askDockerfile(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Validate returns an error if the flag values passed by the user are invalid.
+func (opts *AppInitOpts) Validate() error {
+	if opts.AppType != "" {
+		if err := validateApplicationType(opts.AppType); err != nil {
+			return err
+		}
+	}
+	if opts.AppName != "" {
+		if err := validateApplicationName(opts.AppName); err != nil {
+			return fmt.Errorf("invalid app name %s: %w", opts.AppName, err)
+		}
+	}
+	if opts.DockerfilePath != "" {
+		if _, err := opts.fs.Stat(opts.DockerfilePath); err != nil {
+			return err
+		}
+	}
+	if opts.projectName == "" {
+		return errors.New("no project found, run `project init` first")
+	}
+	return nil
+}
+
+// Execute writes the application's manifest file and stores the application in SSM.
+func (opts *AppInitOpts) Execute() error {
+	manifest, err := manifest.Create(opts.AppName, opts.AppType, opts.DockerfilePath)
+	if err != nil {
+		return fmt.Errorf("failed to generate a manifest %w", err)
+	}
+	manifestBytes, err := manifest.Marshal()
+	if err != nil {
+		return fmt.Errorf("failed to marshal the manifest file %w", err)
+	}
+	manifestPath, err := opts.manifestWriter.WriteManifest(manifestBytes, opts.AppName)
+	if err != nil {
+		return fmt.Errorf("failed to write manifest for app %s: %w", opts.AppName, err)
+	}
+
+	wkdir, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory: %w", err)
+	}
+	relPath, err := filepath.Rel(wkdir, manifestPath)
+	if err != nil {
+		return fmt.Errorf("failed to get relative path of manifest file: %w", err)
+	}
+	log.Infoln()
+	log.Successf("Wrote the manifest for %s app at '%s'\n", color.HighlightUserInput(opts.AppName), color.HighlightResource(relPath))
+	log.Infoln()
+	return nil
+}
+
+func (opts *AppInitOpts) askAppType() error {
+	t, err := opts.prompt.SelectOne(
+		"What type of application do you want to make?",
+		"List of infrastructure patterns.",
```

Comment:
Different applications have different needs. The below patterns/architectures/app types help by setting up all the additional AWS infrastructure needed to get going (LoadBalancers, Cron jobs, etc)